File: backend/models/site.go
package models

import (
    "time"
    "gorm.io/gorm"
)

type Site struct {
    gorm.Model
    Name       string    `json:"name"`
    URL        string    `json:"url" gorm:"unique;not null"`
    Latency    int64     `json:"latency"` 
    IsUp       bool      `json:"is_up"`
    LastStatus int       `json:"last_status"`
    LastCheck  time.Time `json:"last_check"`
}

File: frontend/src/types/site.ts
export interface SiteHistory {
  site_id: number;
  is_up: boolean;
  latency: number;
  checked_at: string;
}

export interface Site {
  ID: number;
  name: string;
  url: string;
  latency: number;
  is_up: boolean;
  last_status: number;
  last_check: string;
  history: SiteHistory[];
}